File: ch8/forkdemo2.c
#include<stdio.h>
#include<stdlib.h>
#include<unistd.h>

int main(int argc, char const *argv[])
{
    printf("my pid is %d\n",getpid());

    fork();
    fork();
    fork();

    printf("my pid is %d\n",getpid());
    return 0;
}

File: ch8/ch8_src/src/bin/forkdemo2.rs
use std::error::Error;

use nix::unistd::{fork, getpid};

// 2^(fork function num)
fn main()->Result<(),Box<dyn Error>>{
    // println!("my pid is {}",getpid());

    unsafe { fork()? };
    unsafe { fork()? };
    unsafe { fork()? };
    
    println!("my pid is {}",getpid());
    Ok(())
}